### PR TITLE
Mutually connected complete functionality

### PR DESCRIPTION
**Pull request by louilinn**
To include in getNetwork result, for an account A

- Those trusted by A
- Those trusting A
- And the accounts who trust someone that A also trusts

in a list of objects for arbitrary account B
```
{
  safeAddress: 0xB (address of B),
  isOutgoing: true if B trusts A,
  isIncoming: true if A trusts B,
  mutualConnections: Array of addresses trusted by A and B
}
```

### CODE REVIEW

**Comment by louilinn** [comment]
Work In Progress: current stuck with loop waiting logic in tests..

File: src/trust.js [hunk 1]
```diff
@@ -92,61 +92,79 @@ export default function createTrustModule(web3, contracts, utils) {
...
-
-            result = Object.values(network);
+                // 3 Add mutual connections
+                const trustersOfTrustee = trustee.user.outgoing;
+                trustersOfTrustee &&
+                  trustersOfTrustee.forEach((trusterOfTrustee) => {
+                    const ttAddress = web3.utils.toChecksumAddress(
+                      trusterOfTrustee.canSendToAddress,
+                    );
+                    if (ttAddress !== trusteeAddress) {
```
Comment (llunaCreixent) [inline src/trust.js:143]: Are the incoming and outgoing datasets from the subgraph include connections to itself?  If so, this `if` is not necessary.

**Review by llunaCreixent — APPROVED** [review]
loving it!

File: src/trust.js (hunk 1 above)
Comment (louilinn) [reply src/trust.js:143]: Yes it was necessary, i did it first without this check

**Pull request merged**